### ISSUE TITLE
cue-lang/setup-cue: intermittent failure to download release manifest results in older version being installed

### ISSUE THREAD

**Issue by jpluscplusm** [cue-lang/cue#2271]
Hello 👋 

When using https://github.com/cue-lang/setup-cue/ and configuring it to install the `latest` version of CUE, the Action intermittently and infrequently sees HTTP request failures when finding out what that `latest` actually is.

As a result, the Action /says/ that it is then installing its "builtin" version 0.4.0:          

```
2023-01-23T14:09:50.0053571Z ##[warning]Cannot get the latest cue releases infos from https://api.github.com/repos/cue-lang/cue/releases. Error HTTPError: Unexpected HTTP response: 403
    at /home/runner/work/_actions/cue-lang/setup-cue/0be332bb74c8a2f07821389447ba3163e2da3bfb/dist/index.js:4630:25
    at Generator.next (<anonymous>)
    at fulfilled (/home/runner/work/_actions/cue-lang/setup-cue/0be332bb74c8a2f07821389447ba3163e2da3bfb/dist/index.js:4540:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  httpStatusCode: 403
}. Using default builtin version v0.4.0.
```

This feels like an arbitrary and old version to potentially end up with -- without notice, and seemingly without a way to guard against. 

Perhaps some of the following approaches could be adopted:

- don't fallback to any version; make the GitHub actions workflow explicitly fail
- keep the "built-in" version of CUE updated, and documented
- put the "is a fallback permitted?" decision in the consumer's hands, via Action config

Personally, I'd prefer the first (noisy explosion) option. It stops masking errors in SaaS tooling I'm using, and gives me a more accurate idea of the error frequency; which I can then either take to my vendor for a fix ... or work out if I'm triggering them, somehow.

**Comment by myitcv** [comment]
> * don't fallback to any version; make the GitHub actions workflow explicitly fail

This feels like the right approach. 

Is the fix in `setup-cue` relatively straightforward?

**Comment by myitcv** [comment]
Marking as v0.6.x - not critical for v0.6.0, but good to get done asap so that people pick up sensible CUE versions when using the action.

**Comment by jpluscplusm** [comment]
**Believed** to be fixed by https://github.com/cue-lang/setup-cue/commit/8e50d5daeea97b99230670818ef0f610522193da.

**Comment by myitcv** [comment]
> **Believed** to be fixed by [cue-lang/setup-cue@8e50d5d](https://github.com/cue-lang/setup-cue/commit/8e50d5daeea97b99230670818ef0f610522193da).

I'll leave you to close this if you feel it's fixed :)

**Comment by jpluscplusm** [comment]
Closing. I've removed the GitHub API from the Action's "latest CUE version" code path.
